src/chain/mod.rs

```diff
@@ -875,6 +1203,50 @@ impl ChainSource {
 
 				Ok(())
 			},
+			Self::Electrum {
+				electrum_runtime_client,
+				fee_estimator,
+				kv_store,
+				logger,
+				node_metrics,
+				..
+			} => {
+				let electrum_client: Arc<ElectrumRuntimeClient> =
+					if let Some(client) = electrum_runtime_client.read().unwrap().as_ref() {
+						Arc::clone(client)
+					} else {
+						debug_assert!(
+							false,
+							"We should have started the chain source before updating fees"
+						);
+						return Err(Error::FeerateEstimationUpdateFailed);
+					};
+
+				let now = Instant::now();
+
+				let new_fee_rate_cache = electrum_client.get_fee_rate_cache_update().await?;
+				fee_estimator.set_fee_rate_cache(new_fee_rate_cache);
+
+				log_info!(
+					logger,
+					"Fee rate cache update finished in {}ms.",
+					now.elapsed().as_millis()
+				);
+
+				let unix_time_secs_opt =
```

Comment:
Another candidate maybe for extraction. Some helper function that updates a timestamp to now and writes it out.